modular/blocksyncer/database/db_statistics.go

```diff
@@ -0,0 +1,62 @@
+package database
+
+import (
+	"context"
+	"time"
+
+	"github.com/bnb-chain/greenfield-storage-provider/pkg/log"
+
+	"github.com/forbole/juno/v4/models"
+	"gorm.io/gorm/clause"
+
+	"github.com/bnb-chain/greenfield-storage-provider/store/bsdb"
+)
+
+func (db *DB) SaveStat(ctx context.Context, stat *models.DataStat) error {
+	return db.Db.Table((&models.DataStat{}).TableName()).Clauses(clause.OnConflict{
+		Columns:   []clause.Column{{Name: "one_row_id"}},
+		UpdateAll: true,
+	}).Create(stat).Error
+}
+
+func (db *DB) GetObjectCount(needSeal bool, blockHeight int64) ([]int64, error) {
+	result := make([]int64, 0, bsdb.ObjectsNumberOfShards)
+	step := int64(1000)
+	for i := 0; i < bsdb.ObjectsNumberOfShards; i++ {
+		sum := int64(0)
+		primaryKey := int64(0)
+		count := int64(0)
+		for {
+			var err error
+			tmpDB := db.Db.Table(bsdb.GetObjectsTableNameByShardNumber(i))
+			if needSeal {
```

Comment:
the condition might miss some updating records.  
What if an old record (the id is small) is updated recently?

Reply:
1. do we have index on update_at ?
2. should we include id range into the condition?